src/api/types.ts

```diff
@@ -143,3 +144,65 @@ export type RawTransactionData = {
 };
 
 export type SwapAPIParams = GetQuoteAPIParams | GetSwapAPIParams;
+
+export type GetTokenDetailsParams = {
+  contractAddress: Address; // The address of the token contract
+  tokenId?: string; // The ID of the token
+};
+
+export type TokenDetails = {
+  name: string;
+  description: string;
+  imageUrl: string;
+  animationUrl: string;
+  mimeType: string;
+  ownerAddress: Address;
+  lastSoldPrice: NFTPrice;
+  contractType: ContractType; // ERC721, ERC1155
+};
+
+export type GetTokenDetailsResponse = TokenDetails | APIError;
+
+export type GetMintDetailsParams = {
+  contractAddress: Address; // The address of the token contract
+  takerAddress?: Address; // The address of the user
+  tokenId?: string; // The ID of the token (required for ERC1155)
+};
+
+export type MintDetails = {
+  name: string;
+  description: string;
+  imageUrl: string;
+  animationUrl: string;
+  mimeType: string;
+  contractType: ContractType;
+  price: NFTPrice;
+  mintFee: NFTPrice;
+  maxMintsPerWallet: number;
+  isEligibleToMint: boolean;
+  creatorAddress: Address;
+  totalTokens: string;
+  totalOwners: string;
+  network: string;
+};
+
+export type GetMintDetailsResponse = MintDetails | APIError;
+
+export type BuildMintTransactionParams = {
+  mintAddress: Address; // The address of the token contract to mint
+  takerAddress: Address; // The address of the user
+  tokenId?: string; // The ID of the token
+  quantity: number; // The number of tokens to mint
+  network?: string; // The network the mint contract is on
+};
+
+type MintTransaction = {
+  call_data: {
+    data: Address;
+    to: Address;
+    from: Address;
+    value: string;
+  };
+};
```

Comment:
nit but can we add descriptions for these

Reply:
updated